### PR TITLE
Hibernate Dao added

### PR DESCRIPTION
**Pull request by joej0e**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            transaction.commit();
+        } catch (Exception e) {
+            logger.error("Can't create Bucket ", e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+    }
+
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/hibernate/BucketDaoHibernateImpl.java]: remove one empty line

File: src/main/java/mate/academy/internetshop/dao/hibernate/OrderDaoHibernateImpl.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            }
+        }
+        return order;
+    }
+
+    @Override
+    public void delete(Long id) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        ;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/hibernate/OrderDaoHibernateImpl.java]: remove

File: src/main/java/mate/academy/internetshop/dao/hibernate/RoleDaoHibernateImpl.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Override
+    public Set<Role> getRoles(Long userId) {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        return session.get(User.class, userId).getRoles();
+    }
+
+}
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/hibernate/RoleDaoHibernateImpl.java:102]: remove empty line

File: src/main/java/mate/academy/internetshop/dao/hibernate/UserDaoHibernateImpl.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    public User getByToken(String token) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("FROM User WHERE token=:token");
+            query.setParameter("token", token);
+            List<User> list = query.list();
+            return list.stream().findFirst().orElseThrow();
+        }
+    }
+}
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/hibernate/UserDaoHibernateImpl.java:132]: :)

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java [hunk 5]
```diff
@@ -28,13 +29,20 @@ public BucketDaoJdbcImpl(Connection connection) {
         super(connection);
     }
 
-    @Override
-    public Bucket create(Bucket bucket) {
+    public Bucket addItem(Long bucketId, Long itemId) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java:32]: clean up all jdbc classes

File: src/main/java/mate/academy/internetshop/model/Bucket.java [hunk 6]
```diff
@@ -1,12 +1,52 @@
...
+            joinColumns = @JoinColumn(name = "bucket_id"),
+            inverseJoinColumns = @JoinColumn(name = "item_id"))
     private List<Item> items;
+
+    @Transient
     private Long userId;
 
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "user_id")
+    @MapsId
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/model/Bucket.java:39]: do we need this annotation?

Comment (joej0e) [reply src/main/java/mate/academy/internetshop/model/Bucket.java:39]: My unidirectional shared primary key working only this way

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**